src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -22,16 +23,18 @@ public class FunctionMetadataManager : IFunctionMetadataManager
         private readonly Lazy<ImmutableArray<FunctionMetadata>> _functionMetadataArray;
         private readonly IOptions<ScriptJobHostOptions> _scriptOptions;
         private readonly IFunctionMetadataProvider _functionMetadataProvider;
+        private readonly IEnumerable<IFunctionProvider> _functionProviders;
         private readonly ILogger _logger;
         private Dictionary<string, ICollection<string>> _functionErrors = new Dictionary<string, ICollection<string>>();
 
-        public FunctionMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IFunctionMetadataProvider functionMetadataProvider, IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        public FunctionMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IFunctionMetadataProvider functionMetadataProvider, IEnumerable<IFunctionProvider> functionProviders, IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
```

Comment:
I think we should be unifying ProxyMetadataManager and your new providers. Consulting the custom providers should be done inside the FunctionMetadataProvider - it should handle proxies and these new custom providers. ProxyMetadataManager should implement your new interface - it already implements the same contract. This way we prove out the extensibility model, which we're already using internally for proxies.

Reply:
I see what you mean. Ah, I just realized that the `FunctionMetada` has `IsProxy` as well. Part of my concern was, if we remove the proxy metadata, how do we know if we need to add proxy descriptor or not. It can check if `FunctionMetada.IsProxy` is true for any.

And then you are saying that `ProxyFunctionDescriptorProvider` would check if the `FunctionMetada` is of `ProxyFunctionMetadata` type, if so, it will get the executor from there and use it? If not, that's an error anyway.

Does that sound accurate?

Reply:
Right